tests/system/HTTP/URITest.php

```diff
@@ -863,56 +863,56 @@ public function testSetBadSegmentSilent()
 
     public function testBasedNoIndex()
     {
-        $this->resetServices();
-
         $_SERVER['HTTP_HOST']   = 'example.com';
         $_SERVER['REQUEST_URI'] = '/ci/v4/controller/method';
 
+        $this->resetServices();
+
         $config            = new App();
-        $config->baseURL   = 'http://example.com/ci/v4';
-        $config->indexPage = 'index.php';
-        $request           = Services::request($config);
-        $request->uri      = new URI('http://example.com/ci/v4/controller/method');
+        $config->baseURL   = 'http://example.com/ci/v4/';
+        $config->indexPage = '';
+        Factories::injectMock('config', 'App', $config);
 
+        $request = Services::request($config);
         Services::injectMock('request', $request);
 
         // going through request
         $this->assertSame('http://example.com/ci/v4/controller/method', (string) $request->getUri());
-        $this->assertSame('/ci/v4/controller/method', $request->getUri()->getPath());
```

Comment:
What causes this? Moving the call to `resetServices()`?

Reply:
No.

`$request->getUri()->getPath()` does not start with `/`. `IncomingRequest::$uri` is a bit different from a normal URI object. The path is changed in the constructor.
https://github.com/codeigniter4/CodeIgniter4/blob/a3eedf4055c1a2895051b654c49311641dd2975f/system/HTTP/IncomingRequest.php#L60-L71

Reply:
Ha yes, that was my note. It's all coming back now.